torch/utils/data/dataset.py

```diff
@@ -293,19 +294,37 @@ def __len__(self):
         return len(self.indices)
 
 
-def random_split(dataset: Dataset[T], lengths: Sequence[int],
+def random_split(dataset: Dataset[T], lengths_or_frac: Sequence[Union[int, float]],
                  generator: Optional[Generator] = default_generator) -> List[Subset[T]]:
     r"""
     Randomly split a dataset into non-overlapping new datasets of given lengths.
+    If a list of fractions is given, the lengths will be computed automatically as floor(frac * len(dataset)).
+    If the sum of the list of the fractions is not equal to 1 or the split leaves a remainder,
+    a new entry will be appended
+    to the end of the list with the remaining fraction
+
     Optionally fix the generator for reproducible results, e.g.:
 
     >>> random_split(range(10), [3, 7], generator=torch.Generator().manual_seed(42))
+    >>> random_split(range(30), [0.3, 0.3, 0.4], generator=torch.Generator().manual_seed(42))
 
     Args:
         dataset (Dataset): Dataset to be split
-        lengths (sequence): lengths of splits to be produced
+        lengths_or_frac (sequence): lengths or fractions of splits to be produced
         generator (Generator): Generator used for the random permutation.
     """
+    if 0 <= sum(lengths_or_frac) <= 1.0:  # type: ignore[arg-type]
```

Comment:
We need to validate for percentages case: Each element should be from 0 to 1

And, I am sorry about my previous wrong statement. We should validate the sum equals to 1.

Reply:
Just realized that I did not actually check if all the elements are positive too.

Reply:
Made changes in the latest commit